Blockzilla/SettingsViewController.swift

```diff
@@ -593,3 +602,74 @@ extension SettingsViewController: SearchSettingsViewControllerDelegate {
         (tableView.cellForRow(at: IndexPath(row: 0, section: 0)) as? SettingsTableViewAccessoryCell)?.accessoryLabelText = engine.name
     }
 }
+
+extension SettingsViewController: INUIAddVoiceShortcutViewControllerDelegate {
+    @available(iOS 12.0, *)
+    func addVoiceShortcutViewController(_ controller: INUIAddVoiceShortcutViewController, didFinishWith voiceShortcut: INVoiceShortcut?, error: Error?) {
+        controller.dismiss(animated: true, completion: nil)
+    }
+    
+    @available(iOS 12.0, *)
+    func addVoiceShortcutViewControllerDidCancel(_ controller: INUIAddVoiceShortcutViewController) {
+        controller.dismiss(animated: true, completion: nil)
+    }
+    
+    @available(iOS 12.0, *)
+    func manageSiri(for activityType: SiriShortcuts.activityType) {
+        INVoiceShortcutCenter.shared.getAllVoiceShortcuts { (voiceShortcuts, error) in
+            if let voiceShortcuts = voiceShortcuts {
```

Comment:
`guard let voiceShortcuts = voiceShortcuts else { return }` to avoid bracing the whole block

Reply:
What thread is `manageSiri` on? If not main, wrap the entire thing in`DispatchQueue.main.async { }`. If those are being used in this func for some other reason than jumping back to main, please note why, thanks.